org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsi/participants/XSIFormatterParticipant.java

```diff
@@ -155,4 +158,104 @@ public String getCurrentLineIndent(XMLBuilder xml, XMLFormattingOptions formatti
+				attr.getNodeAttrName().getEnd() + 2, formattingOptions.getSplitAttributesIndentSize(), startline,
+				normalizedOffset, formattingOptions.isSplitAttributes());
+
+		int lineFeed = split == XSISchemaLocationSplit.onElement ? 1 : 2;
+		int locationNum = 1;
+		for (int i = 0; i < attrValue.length(); i++) {
+			int from = formatterDocument.getLeftWhitespacesOffset(start, start + i + 1);
+			if (Character.isWhitespace(attrValue.charAt(i)) && !Character.isWhitespace(attrValue.charAt(i + 1))
+					&& !StringUtils.isQuote(attrValue.charAt(from - start))) {
+				// Insert newline and indent where required based on setting
+				if (locationNum % lineFeed == 0) {
+					formatterDocument.replaceLeftSpacesWithIndentation(indentLevelOffset, start + i + 1, true, edits);
+				} else {
+					formatterDocument.replaceLeftSpacesWithOneSpace(indentLevelOffset, start + i + 1, edits);
+				}
+				locationNum++;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Returns the offset from opening quote to first non-whitespace character of an
+	 * attribute value
+	 *
+	 * @param originalValue
+	 * @return offset from opening quote to first non-whitespace character of an
+	 *         attribute value
+	 */
+	private static int getFirstContentOffset(String originalValue) {
+		if (originalValue == null) {
+			return -1;
+		}
+		for (int i = 1; i < originalValue.length(); i++) {
+			if (!Character.isWhitespace(originalValue.charAt(i)) && !StringUtils.isQuote(originalValue.charAt(i))) {
+				return i;
+			}
+		}
+		return -1;
+	}
+
+	/**
+	 * Returns indent count of the current line at the beginning of the attribute
+	 * name
+	 * 
+	 * @param tabsize                   the tabsize.
+	 * @param start                     the start offset of the attribute name.
+	 * @param end                       the end offset of the attribute name.
+	 * @param splitAttributesIndentSize the split attribute indent size, 0 if the
+	 *                                  setting is false.
+	 * @param startline                 the offset at the start of the line where
+	 *                                  the attribute name is located
+	 * @param normalizedOffset          the offset at the start of the attribute
+	 *                                  value after normalizing whitespace
+	 * @param isSplitAttributes         the setting of split attribute
+	 * 
+	 * @return indent count of the current line at the beginning of the attribute
+	 *         name.
+	 */
+	private static int getIndentLevelAtAttrName(int tabSize, int start, int end, int splitAttributesIndentSize,
+			int startline, int normalizedOffset, boolean isSplitAttributes) {
+		if (isSplitAttributes) {
+			if ((end - start) % tabSize == 0) {
+				return (end - start) / tabSize + splitAttributesIndentSize;
+			} else {
+				return (end - start) / tabSize + splitAttributesIndentSize + 1;
+			}
+		} else {
+			if ((normalizedOffset - startline) % tabSize == 0) {

```

Comment:
I'm not getting additional spaces to line up the entries exactly anymore, instead it's (neededSpaces/tabWidth) rounded up. Is this intentional? I think it's an okay formatting strategy, but it's different than what was done before, and we will need to adapt the documentation to reflect that.

Reply:
Okay I can change it so that it's the exact number of spaces, I originally thought that we wanted to use tabs.

Reply:
I think the idea is that we use tabs if the formatting settings specify tabs, and spaces if the formatting settings specify spaces. However, we always add a few extra spaces to line it up exactly. At least, this is what the old behaviour was.

Reply:
Indeed, it was the idea. We should have tests with spaces and with tabs settings.